ssl/statem/statem_lib.c

```diff
@@ -46,6 +63,32 @@ int ssl3_do_write(SSL_CONNECTION *s, int type)
     size_t written = 0;
     SSL *ssl = SSL_CONNECTION_GET_SSL(s);
 
+    /*
+     * If we're running the test suite then we may need to mutate the message
+     * we've been asked to write. Does not happen in normal operation.
+     */
+    if (!s->statem.write_in_progress
+            && type == SSL3_RT_HANDSHAKE
+            && s->statem.mutate_handshake_cb != NULL
```

Comment:
Nit: put this condition as first as that should be the most likely condition to fail imo.